c10/core/TensorOptions.h

```diff
@@ -352,6 +352,32 @@ struct C10_API TensorOptions {
     return at::tensorTypeIdToBackend(computeTensorTypeId());
   }
 
+  /// Return the right-biased merge of two TensorOptions.  This has the
+  /// effect of overwriting settings from self with specified options
+  /// of options.
+  ///
+  /// This merging operation respects device merges as specified by
+  /// Device::merge_in, which means it is suitable for implementing
+  /// APIs like new_empty.
+  ///
+  TensorOptions merge_in(TensorOptions options) const noexcept {
```

Comment:
I don't really buy canonicalizing this behavior in TensorOptions.

1) Can't you just get rid of this by not having `device_guard: False`
2) If we want this behavior, we can limit it to tensor_new.  I would NOT expect this is what merge_in does.

Reply:
I think you're right!